haystack/components/preprocessors/document_splitter.py

```diff
@@ -54,40 +52,106 @@ class DocumentSplitter:
 
     def __init__(  # pylint: disable=too-many-positional-arguments
         self,
-        split_by: Literal["function", "page", "passage", "sentence", "word", "line"] = "word",
+        split_by: Literal["function", "page", "passage", "sentence", "word", "line", "nltk_sentence"] = "word",
         split_length: int = 200,
         split_overlap: int = 0,
         split_threshold: int = 0,
         splitting_function: Optional[Callable[[str], List[str]]] = None,
+        respect_sentence_boundary: bool = False,
+        language: Language = "en",
+        use_split_rules: bool = True,
+        extend_abbreviations: bool = True,
     ):
         """
         Initialize DocumentSplitter.
 
-        :param split_by: The unit for splitting your documents. Choose from `word` for splitting by spaces (" "),
-            `sentence` for splitting by periods ("."), `page` for splitting by form feed ("\\f"),
-            `passage` for splitting by double line breaks ("\\n\\n") or `line` for splitting each line ("\\n").
+        :param split_by: The unit for splitting your documents. Choose from:
+            - `word` for splitting by spaces (" ")
+            - `sentence` for splitting by periods (".")
+            - `page` for splitting by form feed ("\\f")
+            - `passage` for splitting by double line breaks ("\\n\\n")
+            - `line` for splitting each line ("\\n")
+            - `nltk_sentence` for splitting by NLTK sentence tokenizer
```

Comment:
I'm not really sure I like how we have to choose `nltk_sentence` here. If anything I'd advocate for the following:
- update "sentence" to mean splitting using NLTK
- add new option "period" to split by period

We are already making a breaking changes so I think we should go ahead and do this to make it more intuitive. Ie if I'm presented with the just options sentence vs nltk_sentence as a user it's really hard to tell what that means.

Reply:
Also if we do this we aren't using the option `sentence` consistently throughout Haystack. Ie the Recursive Chunker has `sentence` but that means nltk and the NLTKDocumentSplitter has `sentence` which also means nltk.

Reply:
agree 👍🏽

Reply:
Awesome! We'd have to follow a similar deprecation notice as @anakin87 suggests for changing this option I would imagine.

Reply:
Hmm I'm not sure, what's your take @anakin87?

Reply:
I don't have strong opinions. And in this case, I think it is OK to be more pragmatic.

Since you are pointing out that the current use of `sentence` across the codebase is inconsistent, we can see this as a bugfix, so no need for deprecations.

What I would do:
- Explicitly point out this change in the release notes.
- Add a warning if using this option, to explicitly explain the change (warning to be removed in a future release).

Agree?

Reply:
> `Add a warning if using this option, to explicitly explain the change (warning to be removed in a future release).`

so this means: create a deprecation notice that `sentence` will be renamed to `period` in the 2.9.0 and only effectively release this new change on `2.10.0` ?

Reply:
No I was proposing to release the change in 2.9.0 but alert users with a warning.